src/Main/main.cpp

```diff
@@ -37,14 +38,21 @@ int main(int argc, char** argv) {
       FOEDAG::Foedag::getGuiType(cmd->WithQt(), cmd->WithQml());
 
   FOEDAG::Compiler* compiler = nullptr;
-  if (cmd->CompilerName() == "openfpga")
-    compiler = new FOEDAG::CompilerOpenFPGA();
-  else
+  FOEDAG::CompilerOpenFPGA* opcompiler = nullptr;
+  if (cmd->CompilerName() == "openfpga") {
+    opcompiler = new FOEDAG::CompilerOpenFPGA();
+    compiler = opcompiler;
+    compiler->SetUseVerific(cmd->UseVerific());
+  } else {
     compiler = new FOEDAG::Compiler();
-  compiler->SetUseVerific(cmd->UseVerific());
+  }
 
   FOEDAG::Foedag* foedag = new FOEDAG::Foedag(
       cmd, mainWindowBuilder, registerAllFoedagCommands, compiler);
-
+  if (opcompiler) {
```

Comment:
Why not pass this values via constructor?

Reply:
See discussion in https://github.com/os-fpga/FOEDAG/pull/314